base/server/upgrade/10.11.0/01-FixChallengePasswordClassPath.py

```diff
@@ -0,0 +1,60 @@
+# Authors:
+#     Christina Fu <cfu@redhat.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright (C) 2015 Red Hat, Inc.
+# All rights reserved.
+#
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import fileinput
+import os
+import re
+import pki.server.upgrade
+
+
+class FixChallengePasswordClassPath(pki.server.upgrade.PKIServerUpgradeScriptlet):
+
+    def __init__(self):
+        super(FixChallengePasswordClassPath, self).__init__()
+        self.message = 'Fix Challenge Password Class Path'
+
+    def upgrade_subsystem(self, instance, subsystem):
+
+        cs_cfg = os.path.join(
+            instance.base_dir,
+            'conf',
+            subsystem.name,
+            'CS.cfg')
+
+        self.backup(cs_cfg)
+
+        for line in fileinput.input(cs_cfg, inplace=1):
+            line = re.sub(r"^oidmap.challenge_password.class=.*",
+                          "oidmap.challenge_password.class=org.mozilla.jss.netscape.security.x509.ChallengePassword",
+                          line)
+
+            line = re.sub(r"^oidmap.extensions_requested_pkcs9.class=.*",
+                          "oidmap.extensions_requested_pkcs9.class=org.mozilla.jss.netscape.security.x509.ExtensionsRequested",
+                          line)
+
+            line = re.sub(r"^oidmap.extensions_requested_vsgn.class=.*",
+                          "oidmap.extensions_requested_vsgn.class=org.mozilla.jss.netscape.security.x509.ExtensionsRequested",
+                          line)
+            print(line, end='')
```

Comment:
IIUC this upgrade script will "fix" the properties in all CS.cfg's instead of "removing" them from non-CA CS.cfg's. I think you'll need to check the `subsystem.name` and remove the properties if it's not CA.

Reply:
@edewata I think @ladycfu said [she'd leave it like it is for now](https://github.com/dogtagpki/pki/pull/3415#discussion_r546161955):

> but on second thought maybe I could just remove them now and if we need it later we could always add.

Reply:
ok thanks!
I also need to address the "line too long" iissue reported by Python lint.

Reply:
I'm tempted to put those lines back into KRA and other's CS.cfg and leave them for later.
I am not sure how to fix "line too long" e.g.
                          "oidmap.challenge_password.class=org.mozilla.jss.netscape.security.x509.ChallengePassword",

Reply:
Long strings can be broken into multiple lines with string concatenations or substitutions. I'll defer to @cipherboy for the ACK.

Reply:
How about:

```
python            line = re.sub(r"^oidmap.challenge_password.class=.*",
                          "oidmap.challenge_password.class=org.mozilla.jss."
                          "netscape.security.x509.ChallengePassword",
                          line)

            line = re.sub(r"^oidmap.extensions_requested_pkcs9.class=.*",
                          "oidmap.extensions_requested_pkcs9.class=org.mozilla.jss."
                          "netscape.security.x509.ExtensionsRequested",
                          line)

            line = re.sub(r"^oidmap.extensions_requested_vsgn.class=.*",
                          "oidmap.extensions_requested_vsgn.class=org.mozilla.jss."
                          "netscape.security.x509.ExtensionsRequested",
                          line)
```